node_modules/gh-admins/tests/test-admins.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2015 "Fronteer LTD"
+ * Grasshopper Event Engine
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var _ = require('lodash');
+var assert = require('assert');
+
+var AuthTestsUtil = require('gh-auth/tests/util');
+var TestsUtil = require('gh-tests/lib/util');
+
+var AdminsTestsUtil = require('./util');
+
+describe('Admins', function() {
+
+    describe('Delete', function() {
+
+        /**
+         * Test that verifies a global admin can delete another global admin
+         */
+        it('verify a global admin can delete another global admin', function(callback) {
+            // Create the global admin rest client to test with
+            TestsUtil.getGlobalAdminRestClient(function(globalAdminClient) {
+
+                // Create a global admin user that we can delete
+                var username = TestsUtil.generateTestUserId();
+                AdminsTestsUtil.assertCreateGlobalAdmin(globalAdminClient, username, username, username, function(createdGlobalAdmin) {
+
+                    // Delete the global admin, ensuring they no longer exist
+                    return AdminsTestsUtil.assertDeleteGlobalAdmin(globalAdminClient, createdGlobalAdmin.id, callback);
+                });
+            });
+        });
+
+        /**
+         * Test that verifies a global admin can delete their own user account
+         */
+        it('verify a global admin can delete their own user account', function(callback) {
+            TestsUtil.getGlobalAdminRestClient(function(globalAdminClient) {
+                var username = TestsUtil.generateTestUserId();
+                AdminsTestsUtil.assertCreateGlobalAdmin(globalAdminClient, username, username, username, function(createdGlobalAdmin) {
+
+                    // Delete the global admin, ensuring they no longer exist
+                    return AdminsTestsUtil.assertDeleteGlobalAdmin(globalAdminClient, createdGlobalAdmin.id, callback);
```

Comment:
It would probably make sense to make it so that a session that is for a user id that doesn't exist is equivalent to an anonymous session.